logfire/propagate.py

```diff
@@ -70,3 +71,26 @@ def attach_context(carrier: ContextCarrier) -> Iterator[None]:
         yield
     finally:
         context.attach(old_context)
+
+
+def build_span_link(carrier: ContextCarrier) -> Link:
+    """Build a span link from a context carrier.
+
+    This is useful when you want to link a span in a different service to the current span.
+
+    ```py
+    from logfire.propagate import build_span_link, get_context
+
+    logfire_context = get_context()
+
+    ...
+
+    # later on in another thread, process or service
+    link = build_span_link(logfire_context)
```

Comment:
Why wouldn't the user use `with attach_context` here?

Reply:
If it's attached, then it doesn't make sense to add a span link, does it?

Reply:
Right, I'm questioning adding (and recommending) `build_span_link`. In this situation it seems more sensible to use a parent-child relationship instead of a span link.

Reply:
I mean, if I change the comment to "later on a task queue", would that be better? Or what you suggest instead?

By the way, I'm not sure `build_span_link` is the best API for this, if we `build_span_context` also works. I don't care which if the `span(..., _links=[...])` accepts one of them.

Reply:
> I mean, if I change the comment to "later on a task queue", would that be better?

No because regular distributed tracing still seems better.

> Or what you suggest instead?

I honestly don't know when span links are useful in general. I suppose in your bigger example there's already a parent `process_message`. If we assume that for whatever reason we really want to keep that as the parent, then we have to use a link. But then the docs should clarify that so that users only use links when it makes sense.

> By the way, I'm not sure `build_span_link` is the best API for this, if we `build_span_context` also works. I don't care which if the `span(..., _links=[...])` accepts one of them.

I like that idea. Maybe name it `get_span_context`.

Reply:
> No because regular distributed tracing still seems better.

You disagree with https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3002 ?

> I honestly don't know when span links are useful in general.

Besides Airflow, I'm not sure who uses them... But I feel they are **supposed** to be useful when you "trigger" something e.g. you have an endpoint, and you trigger a veryyyy long task somewhere else - you want to be able to close the span from the endpoint, and have a link to this very long task.

> I like that idea. Maybe name it get_span_context.

👍

Reply:
> > No because regular distributed tracing still seems better.
> 
> You disagree with [open-telemetry/opentelemetry-python-contrib#3002](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3002) ?

Overall, yes. What made you post that?

> > I honestly don't know when span links are useful in general.
> 
> Besides Airflow, I'm not sure who uses them... But I feel they are **supposed** to be useful when you "trigger" something e.g. you have an endpoint, and you trigger a veryyyy long task somewhere else - you want to be able to close the span from the endpoint,

The endpoint span will still be closed. It'll just have children that starts after it ends, which feels weird but should be harmless.

> and have a link to this very long task.

The endpoint span won't have a link to the task span, only the other way around. Someone looking at the endpoint span has no easy way to find the task span.